fe/fe-core/src/main/java/com/starrocks/qe/scheduler/QueryRuntimeProfile.java

```diff
@@ -256,7 +258,7 @@ public boolean isFinished() {
     }
 
     public boolean addListener(Consumer<Boolean> task) {
-        if (EXECUTOR.getQueue().size() > Config.profile_process_blocking_queue_size) {
+        if (EXECUTOR.getQueue().remainingCapacity() <= 0) {
```

Comment:
so only 'corePoolSize' threads will be created? because when  queue is full ,we change to sync profile mode

Reply:
not really